contracts/strategies/traderjoe/TraderJoe.sol

```diff
@@ -84,62 +171,76 @@ contract TraderJoe is UUPSUpgradeable, StrategyOwnablePausableBaseUpgradeable {
         TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
             .getStorage();
 
-        uint256 pairDepositTokenBalanceBeforeSwap = strategyStorage
-            .pairDepositToken
-            .balanceOf(address(this));
-
-        // swapping depositToken to pairDepositToken
+        // Swap half of depositToken to pairDepositToken.
         uint256 swapAmount = amount / 2;
```

Comment:
What happens if I provide liquidity to a bin that is AssetX, but I only provide AssetY:
1. will V2 itself convert it OR
2. it just doesn't take AssetY

Reply:
If it's active one, it takes AssetY and impose swap fee. If it's inactive one, it reverts.